circonusllhist.go

```diff
@@ -817,6 +817,35 @@ func (h *Histogram) CopyAndReset() *Histogram {
 	}
 	return newhist
 }
+
+// Copy creates and returns an exact copy of a histogram without
+// resetting the contents of the original.
+func (h *Histogram) Copy() *Histogram {
+	if h.useLocks {
+		h.mutex.Lock()
+		defer h.mutex.Unlock()
+	}
+
+	bvs := []bin{}
+	for _, v := range h.bvs {
+		bvs = append(bvs, v)
+	}
+
+	lookup := [256][]uint16{}
+	for i, u := range h.lookup {
+		for _, v := range u {
```

Comment:
What happens when u is nil?  Is that possible?

Reply:
The range keyword handles the nil check.  The loop will end immediately if u is nil.